src/prefect/utilities/context.py

```diff
@@ -93,7 +98,11 @@ def __call__(self, *args: MutableMapping, **kwargs: Any) -> Iterator["Context"]:
         """
         previous_context = self.copy()
         try:
-            self.update(*args, **kwargs)
+            new_context = dict(*args, **kwargs)
+            if "config" in new_context:
+                new_config = merge_dicts(self.get("config", {}), new_context["config"])
+                new_context["config"] = as_nested_dict(new_config, dct_class=Config)
```

Comment:
This is actually a really subtle detail -- but if a value in the config were itself a dictionary, this would replace that dictionary with a `Config`. Now the config really shouldn't have any dictionaries in it, as its TOML defined and the only possible dictionaries are keys, not values, but it's just something to call out.

Reply:
Yea, I thought about this some; because we don't hedge against it in our normal config, I wasn't concerned here.  Ultimately I'd argue users should _only_ use config by setting a finite list of environment variables which act as levers for affecting cloud functionality (or local flow-building functionality, i.e., default settings).  For all other dynamic values, users can always provide `context` when creating a flow run (whether locally or in Cloud).

Reply:
Yeah I agree with you.